File: Baekjoon/11657_타임머신/B_11657.py
import sys


def bellman_ford():
    for i in range(N - 1):
        for j in range(M):
            start = routes[j][0]
            end = routes[j][1]
            value = routes[j][2]

            if distance[start] != INF and distance[end] > distance[start] + value:
                distance[end] = distance[start] + value

    for i in range(1):
        for j in range(M):
            start = routes[j][0]
            end = routes[j][1]
            value = routes[j][2]

            if distance[start] != INF and distance[end] > distance[start] + value:
                return False

    return True


# 정점, 간선
N, M = map(int, sys.stdin.readline().split())
routes = []
for _ in range(M):
    A, B, C = map(int, sys.stdin.readline().split())
    routes.append([A, B, C])

INF = float('inf')
distance = [INF] * (N + 1)
distance[1] = 0
answer = bellman_ford()

if not answer:
    sys.stdout.write(str(-1) + "\n")
else:
    for k in range(2, N + 1):
        # 갈 수 없는 루트도 존재할 수 있다
        if distance[k] == INF:
            sys.stdout.write(str(-1) + "\n")
        else:
            sys.stdout.write(str(distance[k]) + "\n")

File: Baekjoon/11657_타임머신/B_11657.js
const bellmanFord = function() {
    for (let i = 0; i < M - 1; i++) {
        for (let j = 0; j < M; j++) {
            const start = routes[j][0]
            const end = routes[j][1]
            const value = routes[j][2]

            if (distance[start] != Infinity && distance[end] > distance[start] + value) {
                distance[end] = distance[start] + value
            }
        }
    }

    for (let i = 0; i < 1; i++) {
        for (let j = 0; j < M; j++) {
            const start = routes[j][0]
            const end = routes[j][1]
            const value = routes[j][2]

            if (distance[start] != Infinity && distance[end] > distance[start] + value) {
                return false
            }
        }
    }

    return true
}

const fs = require('fs')
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n')

const NM = input[0].split(' ').map(v => parseInt(v))
const N = NM[0]
const M = NM[1]

let routes = []
for (let i = 1; i <= M; i++) {
    const route = input[i].split(' ').map(v => parseInt(v))
    const s = route[0]
    routes.push(route)
}

let distance = new Array(N + 1).fill(Infinity)
distance[1] = 0

const val = bellmanFord()
if (val) {
    for (let i = 2; i <= N; i++) {
        if (distance[i] != Infinity) {
            console.log(distance[i])
        } else {
            console.log(-1)
        }
    }
} else {
    console.log(-1)
}
